Mathlib/LinearAlgebra/TensorProduct/RightExactness.lean

```diff
@@ -361,3 +392,210 @@ theorem TensorProduct.map_ker :
+        | zero =>
+          use 0
+          simp only [map_zero, smul_eq_mul, mul_zero]
+        | tmul x y =>
+          use (a * x) ⊗ₜ[R] (b •y)
+          simp only [LinearMap.lTensor_tmul, Submodule.coeSubtype, smul_eq_mul, tmul_mul_tmul]
+          rfl
+        | add x y hx hy =>
+          obtain ⟨x', hx'⟩ := hx
+          obtain ⟨y', hy'⟩ := hy
+          use x' + y'
+          simp only [map_add, hx', smul_add, hy']
+      | add a b ha hb =>
+        obtain ⟨x', ha'⟩ := ha
+        obtain ⟨y', hb'⟩ := hb
+        use x' + y'
+        simp only [map_add, ha', add_smul, hb']
+
+  · rintro x ⟨y, rfl⟩
+    induction y using TensorProduct.induction_on with
+    | zero =>
+        rw [map_zero]
+        apply zero_mem
+    | tmul a b =>
+        simp only [LinearMap.lTensor_tmul, Submodule.coeSubtype]
+        suffices : a ⊗ₜ[R] (b : B) = (a ⊗ₜ[R] (1 : B)) * ((1 : A) ⊗ₜ[R] (b : B))
+        rw [this]
+        simp only [AddSubsemigroup.mem_carrier, AddSubmonoid.mem_toSubsemigroup,
+          Submodule.mem_toAddSubmonoid, Submodule.restrictScalars_mem]
+        apply Ideal.mul_mem_left
+        apply Ideal.mem_map_of_mem
+        exact Submodule.coe_mem b
+        simp only [Submodule.coe_restrictScalars, Algebra.TensorProduct.tmul_mul_tmul,
+          mul_one, one_mul]
+    | add x y hx hy =>
+        rw [map_add]
+        apply Submodule.add_mem _ hx hy
+
+-- Now, we can prove the right exactness properties of the tensor product,
+-- in its versions for algebras
+
+variable {R : Type*} [CommRing R]
+  {A B C D : Type*} [Ring A] [Ring B] [Ring C] [Ring D]
+  [Algebra R A] [Algebra R B] [Algebra R C] [Algebra R D]
+  (f : A →ₐ[R] B) (g : C →ₐ[R] D)
+
+/-- If `g` is surjective, then the kernel of `(id A) ⊗ g` is generated by the kernel of `g` -/
+lemma Algebra.TensorProduct.lTensor_ker (hg : Function.Surjective g) :
+    RingHom.ker (map (AlgHom.id R A) g) =
+      (RingHom.ker g).map (Algebra.TensorProduct.includeRight : C →ₐ[R] A ⊗[R] C) := by
+  rw [← Submodule.restrictScalars_inj R]
+  have : (RingHom.ker (map (AlgHom.id R A) g)).restrictScalars R =
+    LinearMap.ker (LinearMap.lTensor A (AlgHom.toLinearMap g)) := rfl
+  rw [this, Ideal.map_includeRight_eq]
+  rw [(lTensor_exact A g.toLinearMap.exact_subtype_ker_map hg).linearMap_ker_eq]
+  rfl
+
+/-- If `f` is surjective, then the kernel of `f ⊗ (id B)` is generated by the kernel of `f` -/
+lemma Algebra.TensorProduct.rTensor_ker (hf : Function.Surjective f) :
+    RingHom.ker (map f (AlgHom.id R C)) =
+      (RingHom.ker f).map (Algebra.TensorProduct.includeLeft : A →ₐ[R] A ⊗[R] C) := by
+  rw [← Submodule.restrictScalars_inj R]
+  have : (RingHom.ker (map f (AlgHom.id R C))).restrictScalars R =
+    LinearMap.ker (LinearMap.rTensor C (AlgHom.toLinearMap f)) := rfl
+  rw [this, Ideal.map_includeLeft_eq]
+  rw [(rTensor_exact C f.toLinearMap.exact_subtype_ker_map hf).linearMap_ker_eq]
+  rfl
+
+lemma AlgHom.coe_ker {R A B : Type*} [CommSemiring R] [Semiring A] [Semiring B]
```

Comment:
Here and in the following lemmas you can use the variables defined above, right?

Reply:
yes, and while we're there, why not transferring this lemma to the adequate file?

Reply:
in fact, not quite : in that lemma, R is a CommSemiring, not a CommRing, and similarly for A, B.

Reply:
Ah yes, of course! Thanks.

Reply:
I moved it to the end of `RingTheory.Ideal.Operations`, where docs#find_home told me I could put it. It's where docs#RingHom.ker is defined.